src/base/settingsstorage.h

```diff
@@ -62,4 +62,27 @@ public slots:
     mutable QReadWriteLock m_lock;
 };
 
+template<typename T>
+class SettingsStorageHelper
+{
+public:
+    SettingsStorageHelper(const QString &keyName)
+        : m_keyName(keyName)
+    {
+    }
+
+    void set(const T &value)
+    {
+        SettingsStorage::instance()->storeValue(m_keyName, value);
+    }
+
+    T get(const QVariant &defaultValue = QVariant())
```

Comment:
Can't understand how the default parameter value is supposed to work: can you call `QVariant::value<T>()` for an empty `QVariant`? What will it return? Would it be better to rephrase it in the following way:
```C++
T get (const T &defauleValue = T())
```